library/SimplePie/Locator.php

```diff
@@ -75,12 +75,19 @@ public function __construct(SimplePie_File $file, $timeout = 10, $useragent = nu
 		$this->force_fsockopen = $force_fsockopen;
 		$this->curl_options = $curl_options;
 
-		if (class_exists('DOMDocument'))
+		if (class_exists('DOMDocument') && $this->file->body != '')
```

Comment:
Wouldn't it be better to use the [`empty()`](https://www.php.net/manual/en/function.empty.php) function? Can it be `null`?

Reply:
`loadHTML()` will raise an error if the input is null or empty. This test covers both.

Reply:
Oh yeh, "type juggling".